src/main/java/com/sombra/edu/springmenteeproject/service/impl/WalletServiceImpl.java

```diff
@@ -1,73 +1,62 @@
 package com.sombra.edu.springmenteeproject.service.impl;
 
-import com.sombra.edu.springmenteeproject.entity.UserAccount;
 import com.sombra.edu.springmenteeproject.entity.Wallet;
-import com.sombra.edu.springmenteeproject.exception.NotFoundException;
-import com.sombra.edu.springmenteeproject.exception.NullEntityReferenceException;
 import com.sombra.edu.springmenteeproject.repository.WalletRepository;
 import com.sombra.edu.springmenteeproject.service.WalletService;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
 
 @Service
 @Slf4j
 public class WalletServiceImpl implements WalletService {
 
-    private final WalletRepository repository;
-
+    private final WalletRepository walletRepository;
     @Autowired
-    public WalletServiceImpl(WalletRepository repository) {
-        this.repository = repository;
+    public WalletServiceImpl(WalletRepository walletRepository) {
+        this.walletRepository = walletRepository;
     }
 
     @Override
-    public Optional<Wallet> createNewWallet(Wallet wallet) {
-        if (repository.findWalletById(wallet.getId()).isEmpty()) {
-            return repository.saveWallet(wallet);
-        } else {
+    public Wallet createNewWallet(Wallet wallet) {
+        if (walletRepository.existsById(wallet.getId())) {
             log.warn("Element is already exist");
-            throw new NotFoundException("Element is already exist");
+            throw new IllegalArgumentException("Element is already exist");
         }
+        return walletRepository.save(wallet);
     }
 
     @Override
-    public Optional<Wallet> editWallet(Wallet wallet) {
-        if (repository.findWalletById(wallet.getId()).isPresent()) {
-            return repository.updateWallet(wallet);
-        } else {
+    public Wallet editWallet(Wallet wallet) {
+        if (!walletRepository.existsById(wallet.getId())) {
             log.warn("Can't find the element");
-            throw new NotFoundException("Can't find the element");
+            throw new IllegalArgumentException("Can't find the element");
```

Comment:
Why change from `NotFoundException` to `IllegalArgumentException`? the first approach was correct